uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -53,8 +53,34 @@ define([
          * @return {Promise} Resolved with {Object} Updated Leave request
          */
         update: function (attributes) {
-          return LeaveRequestAPI.sendPOST('LeaveRequest', 'create', _.assign({}, this.attributes(), attributes))
+          var updatedAttributes = _.assign({}, this.attributes(), attributes);
```

Comment:
I just realized that this is not how `update()` is supposed to work. It shouldn't accept any parameter, it should just send the internal state of the instance to the api
```js
update: function () {
  return LeaveRequestAPI.update(this.attributes);
});
```
Usage example:
```js
var instance = LeaveRequest.find(123);

instance.status_id; // 1
instance.status_id = 2;

instance.update().then(function () {
  instance.status_id; // 2
})
.catch(function (err) {

});
```